tests/test_inventory_orphaned_workflows.py

```
@@ -0,0 +1,699 @@
+        inventory.parse_link_has_next("")
+    with pytest.raises(inventory.InventoryError, match="malformed"):
+        inventory.parse_link_has_next(1)
+
+
+def test_decode_registry_path_rejects_encoding_and_traversal() -> None:
+    """Percent-encoding, NULs, backslashes, and `..` fail closed."""
+    assert (
+        inventory.decode_registry_path(".github/workflows/ci.yml")
+        == ".github/workflows/ci.yml"
+    )
+    with pytest.raises(inventory.InventoryError, match="NUL|missing"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)